### PR TITLE
feat: added lifecycle to node_count

### PR DESCRIPTION
**Pull request by mentlak0**
Change to help ignore changes by detecting difference in current settings to match config.

### CODE REVIEW

**Comment by ankitm123** [comment]
```
<html>
<body>
<!--StartFragment-->

Your version of Terraform is out of date! The latest version
--
  |   |   |   | ╷
  |   |   |   | │ Error: Unsupported attribute
  |   |   |   | │
  |   |   |   | │   on terraform-jx-cluster-aks/cluster/main.tf line 21, in resource "azurerm_kubernetes_cluster" "aks":
  |   |   |   | │   21:       ignore_changes = [node_count]
  |   |   |   | │
  |   |   |   | │ This object has no argument, nested block, or exported attribute named
  |   |   |   | │ "node_count".

<!--EndFragment-->
</body>
</html>
```

**Comment by tomhobson** [comment]
Ok we'll pull in the updated version and can merge it in this PR

**Comment by ankitm123** [comment]
Dont think it's a tf issue, is it supported in the latest provider version? Or may be it's not the right place to use node_count? may be in node pool?

File: terraform-jx-cluster-aks/cluster/main.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@ resource "azurerm_kubernetes_cluster" "aks" {
     orchestrator_version = var.cluster_version
     enable_auto_scaling  = var.max_node_count == null ? false : true
   }
-
+
```
Comment (tomhobson) [inline terraform-jx-cluster-aks/cluster/main.tf:19]: Remove this whitespace

**Review by tomhobson — CHANGES_REQUESTED** [review]

**Pull request closed without merging**